lib/core/state/replication.js

```diff
@@ -0,0 +1,46 @@
+const MongoClient = require('../clients/mongodb-client')
+const ReplicationCommands = require('../commands/replication-commands')
+const { REPLICATION_STATUS, MongoServerState } = require('./mongo-server')
+const logger = require('../../helpers/logger')
+
+const REPLICA_SET_STATUS = {
+  UNKNOWN: 0,
+  UNINITIALIZED: 1
+}
+
+class ReplicaSetState {
+  constructor(configuration) {
+    this.hostnames = configuration.hostnames
+    this.replicaSetName = configuration.replicaSetName
+    this.ClientClass = configuration.ClientClass || MongoClient
```

Comment:
lower case?  its a variable not a actual class.

Isn't this supposed to also take in our stubs/real client?

Reply:
Change this, so any client can be injected